detect_secrets/core/audit.py

```diff
@@ -109,11 +194,128 @@ def _secret_generator(baseline):
             yield filename, secret
 
 
+def _get_secrets_to_compare(old_baseline, new_baseline):
+    """
+    :rtype: list(tuple)
+    :param: tuple is in the following format:
+        filename: str; filename where identified secret is found
+        secret: dict; PotentialSecret json representation
+        is_secret_removed: bool; has the secret been removed from the
+            new baseline?
+    """
+    def _check_string(a, b):
+        if a == b:
+            return 0
+        if a < b:
+            return -1
+        return 1
+
+    def _check_secret(a, b):
+        if a == b:
+            return 0
+
+        if a['line_number'] < b['line_number']:
+            return -1
+        elif a['line_number'] > b['line_number']:
+            return 1
+
+        return _check_string(a['hashed_secret'], b['hashed_secret'])
```

Comment:
As a reminder of context: We [punted off](https://github.com/Yelp/detect-secrets/pull/76) changing the audit functionality to de-duplicate secrets with the same `hashed_secret`, and accepted larger baselines.

However this brings up something interesting, if I merged in the [current Keyword plugin PR](https://github.com/Yelp/detect-secrets/pull/86), and the Keyword detector found something of the same hash (new baseline) as a Hex High Entropy plugin in the old baseline, either (a) we would say that the Keyword Detector secret got added (and that wouldn't mean we found a new secret) or (b) we would say that the Hex High Entropy plugin got added, (assuming the Hex High Entropy secret was in both baseline) which is definitely not True.